rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/core/ReconnectableRSocket.kt

```diff
@@ -97,30 +98,33 @@ private class ReconnectableRSocket(
     private val state: StateFlow<ReconnectState>,
 ) : RSocket {
 
-    private val reconnectHandler = state.mapNotNull { it.handleState { null } }.take(1)
+    private val reconnectHandler = state.mapNotNull { it.current() }.take(1)
 
-    //null pointer will never happen
-    private suspend fun currentRSocket(): RSocket = state.value.handleState { reconnectHandler.first() }!!
+    private suspend fun currentRSocket(closeable: Closeable): RSocket = closeable.closeOnError { currentRSocket() }
```

Comment:
It makes sense after digging into the code, but the closeOnError confused me for a bit.  I think wildcard imports don't help.